tests/python/pytorch/graphbolt/test_csc_sampling_graph.py

```diff
@@ -383,3 +383,117 @@ def test_in_subgraph_heterogeneous():
     assert torch.equal(
         in_subgraph.type_per_edge, torch.LongTensor([2, 2, 1, 3, 1, 3, 3])
     )
+
+
+def check_tensors_on_the_same_shared_memory(t: torch.Tensor, t2: torch.Tensor):
+    """Check if two tensors are on the same shared memory.
+
+    This function copies a random tensor to `t` and checks whether `t2` has the
+    same random value. It cannot use `.data_ptr()` because the memory pointers
+    created by different `shm_open` calls are different.
```

Comment:
This function copies a random tensor value to `t1` and checks whether `t2` holds the same random value and checks whether `t2` is a distinct tensor object from `t1`. Their equality confirms that they are separate tensors that rely on the shared memory for their tensor value.